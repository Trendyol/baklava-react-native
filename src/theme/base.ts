export const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];

export const fonts = ['System', 'System', 'System', 'System', 'System'];

export const fontSizes = [12, 14, 16, 20, 24, 28, 32];

export const fontWeights = ['300', '400', '500', '600', '700'];

export const lineHeights = [12, 14, 16, 20, 24, 28, 32, 36];

export const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16,
};

export const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
};

export default {
  space,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  colors,
};
