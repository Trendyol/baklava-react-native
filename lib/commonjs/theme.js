"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.space = exports.radii = exports.fontSizes = exports.default = exports.colors = void 0;
const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];
exports.space = space;
const fontSizes = [12, 14, 16, 20, 24, 28, 32];
exports.fontSizes = fontSizes;
const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16
};
exports.radii = radii;
const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
  // Gradient
  gradientPurple: ['#8C4EFF', '#5794FF'],
  gradientOrange: ['#FFA964', '#F27A1A']
};
exports.colors = colors;
var _default = {
  space,
  fontSizes,
  radii,
  colors
};
exports.default = _default;
//# sourceMappingURL=theme.js.map