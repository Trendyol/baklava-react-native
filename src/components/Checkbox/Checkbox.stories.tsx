import { ComponentMeta, ComponentStory } from '@storybook/react-native';
import React from 'react';
import Box from '../Box/Box';
import Text from '../Text/Text';
import Checkbox from './Checkbox';

const CheckboxMeta: ComponentMeta<typeof Checkbox> = {
  title: 'Checkbox',
  component: Checkbox,
  argTypes: {
    onPress: { action: 'pressed the button' },
  },
  args: {
    checked: true,
    indeterminate: false,
    disabled: false,
  },
};

export default CheckboxMeta;

type CheckboxStory = ComponentStory<typeof Checkbox>;

export const Basic: CheckboxStory = args => {
  const [checked, setChecked] = React.useState(false);
  React.useEffect(() => {
    setChecked(args.checked);
  }, [args.checked]);
  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        Checkbox
      </Text>
      <Checkbox
        {...args}
        onPress={() => setChecked(!checked)}
        checked={checked}
        indeterminate={args.indeterminate}
        label="lorem ipsum"
        m="2xs"
      />
    </>
  );
};

export const Cases: CheckboxStory = () => {
  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        Checkbox
      </Text>
      <Box px="m" py="2xs" flexDirection="row">
        <Checkbox checked={false} />
        <Box width={8} />
        <Checkbox checked={true} indeterminate={false} />
        <Box width={8} />
        <Checkbox checked={true} indeterminate={true} />
        <Box width={8} />
        <Checkbox disabled checked={false} />
        <Box width={8} />
        <Checkbox disabled checked={true} indeterminate={false} />
        <Box width={8} />
        <Checkbox disabled checked={true} indeterminate={true} />
      </Box>

      <Box px="m" py="2xs">
        <Box height={12} />
        <Checkbox checked={false} label="checked: false" />
        <Box height={12} />
        <Checkbox
          checked={false}
          indeterminate={false}
          label="checked: false, indeterminate: false"
        />
        <Box height={12} />
        <Checkbox
          checked={false}
          indeterminate={true}
          label="checked: false, indeterminate: true"
        />
        <Box height={12} />
        <Checkbox checked={true} label="checked: true" />
        <Box height={12} />
        <Checkbox
          checked={true}
          indeterminate={false}
          label="checked: true, indeterminate: false"
        />
        <Box height={12} />
        <Checkbox
          checked={true}
          indeterminate={true}
          label="checked: true, indeterminate: true"
        />
      </Box>

      <Box px="m" py="xs">
        <Text py="xs" variant="subtitle02Medium">
          Disable:
        </Text>
        <Checkbox
          disabled={true}
          checked={false}
          label="disabled, checked: false"
        />
        <Box height={12} />
        <Checkbox
          disabled={true}
          checked={false}
          indeterminate={false}
          label="disabled, checked: false, indeterminate: false"
        />
        <Box height={12} />
        <Checkbox
          disabled={true}
          checked={false}
          indeterminate={true}
          label="disabled, checked: false, indeterminate: true"
        />
        <Box height={12} />
        <Checkbox
          disabled={true}
          checked={true}
          label="disabled, checked: true"
        />
        <Box height={12} />
        <Checkbox
          disabled={true}
          checked={true}
          indeterminate={false}
          label="disabled, checked: true, indeterminate: false"
        />
        <Box height={12} />
        <Checkbox
          disabled={true}
          checked={true}
          indeterminate={true}
          label="disabled, checked: true, indeterminate: true"
        />
      </Box>

      <Box px="m" py="xs">
        <Text py="xs" variant="subtitle02Medium">
          Alignment:
        </Text>
        <Checkbox
          checked={false}
          label="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        />
        <Box height={12} />
        <Checkbox
          checked={false}
          indeterminate={false}
          alignItems="center"
          label="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        />
      </Box>
    </>
  );
};

export const CheckboxWithIcon: CheckboxStory = () => {
  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        Checkbox With Icon
      </Text>

      <Box px="m" py="2xs">
        <Box height={12} />
        <Checkbox
          checked={false}
          label="checked: false, icon: heart"
          icon="heart"
        />
        <Box height={12} />
        <Checkbox
          checked={true}
          label="checked: true, icon: heart"
          icon="heart"
        />
        <Box height={12} />
        <Checkbox
          checked={true}
          indeterminate={true}
          label="checked: true, indeterminate: true, icon: heart"
          icon="heart"
        />
        <Box height={12} />
        <Checkbox
          disabled={true}
          checked={false}
          label="checked: false, disabled, icon: heart"
          icon="heart"
        />
      </Box>
    </>
  );
};
