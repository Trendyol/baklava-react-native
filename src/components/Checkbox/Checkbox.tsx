import React from 'react';
import {
  LayoutProps,
  SpacingProps,
  SpacingShorthandProps,
  VariantProps,
  createRestyleComponent,
  createVariant,
  layout,
  spacing,
  spacingShorthand,
} from '@ergenekonyigit/restyle';
import { TouchableOpacity } from 'react-native';
import { Theme } from '../../theme';
import Box from '../Box/Box';
import Icon from '../Icon/Icon';
import Text from '../Text/Text';

const variant = createVariant({ themeKey: 'checkboxVariants' });

type TickboxProps = VariantProps<Theme, 'checkboxVariants'> &
  React.ComponentProps<typeof Box>;

const Tickbox = createRestyleComponent<TickboxProps, Theme>([variant], Box);

type ContainerProps = React.ComponentProps<typeof TouchableOpacity> &
  LayoutProps<Theme> &
  SpacingProps<Theme> &
  SpacingShorthandProps<Theme>;

const CheckboxContainer = createRestyleComponent<ContainerProps, Theme>(
  [layout, spacing, spacingShorthand],
  TouchableOpacity,
);

type CheckboxProps = React.ComponentProps<typeof CheckboxContainer> & {
  checked: boolean;
  indeterminate?: boolean;
  label?: string;
  disabled?: boolean;
  onPress?: () => void;
};

const Checkbox = ({
  checked,
  indeterminate,
  label,
  disabled = false,
  onPress,
  ...rest
}: CheckboxProps) => {
  const icon = React.useMemo(() => {
    if (!checked) {
      return null;
    }
    if (indeterminate) {
      return (
        <Icon
          size="4xs"
          name="minus"
          color={disabled ? 'contentTertiary' : 'white'}
          testID="minus-icon"
        />
      );
    }
    return (
      <Icon
        size="3xs"
        name="check"
        color={disabled ? 'contentTertiary' : 'white'}
        testID="check-icon"
      />
    );
  }, [checked, indeterminate, disabled]);

  return (
    <CheckboxContainer
      flexDirection="row"
      alignSelf="flex-start"
      disabled={disabled}
      onPress={onPress}
      {...rest}>
      <Tickbox
        testID="tickbox"
        variant={
          checked
            ? disabled
              ? 'disabledChecked'
              : 'checked'
            : disabled
            ? 'disabledUnchecked'
            : 'unchecked'
        }>
        {icon}
      </Tickbox>
      {label ? (
        <Text
          testID="label"
          marginLeft="2xs"
          color={
            disabled
              ? 'contentPassive'
              : checked
              ? 'primaryColor'
              : 'contentPrimary'
          }>
          {label}
        </Text>
      ) : null}
    </CheckboxContainer>
  );
};

export default Checkbox;
