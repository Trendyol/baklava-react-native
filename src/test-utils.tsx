import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
