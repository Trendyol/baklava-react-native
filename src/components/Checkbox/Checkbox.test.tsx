import React from 'react';
import theme from '../../theme';
import { fireEvent, render } from '../../test-utils';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
  beforeEach(() => {});

  test('should render Checkbox correctly when checked is false', () => {
    // when
    const { toJSON, getByTestId } = render(<Checkbox checked={false} />);
    const checkboxComponent = getByTestId('tickbox');

    // then
    expect(toJSON()).toMatchSnapshot();
    expect(checkboxComponent.props.style[1][0].borderColor).toBe(
      theme.colors.neutralLighter,
    );
    expect(checkboxComponent.props.style[1][0].backgroundColor).toBe(
      theme.colors.neutralFull,
    );
  });

  test('should render Checkbox correctly when checked is true', () => {
    // when
    const { toJSON, getByTestId } = render(<Checkbox checked={true} />);
    const checkboxComponent = getByTestId('tickbox');

    // then
    expect(toJSON()).toMatchSnapshot();
    expect(checkboxComponent.props.style[1][0].borderColor).toBe(
      theme.colors.primaryKey,
    );
    expect(checkboxComponent.props.style[1][0].backgroundColor).toBe(
      theme.colors.primaryKey,
    );
  });

  test('should render Checkbox correctly when checked is true, indeterminate is false', () => {
    // when
    const { toJSON, getByTestId } = render(
      <Checkbox checked={true} indeterminate={false} />,
    );
    const checkboxComponent = getByTestId('tickbox');

    // then
    expect(toJSON()).toMatchSnapshot();
    expect(checkboxComponent.props.style[1][0].borderColor).toBe(
      theme.colors.primaryKey,
    );
    expect(checkboxComponent.props.style[1][0].backgroundColor).toBe(
      theme.colors.primaryKey,
    );
  });

  test('should render Checkbox correctly when checked is true, indeterminate is true', () => {
    // when
    const { toJSON } = render(<Checkbox checked={true} indeterminate={true} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render Checkbox correctly when disable, checked is false', () => {
    // when
    const { toJSON } = render(<Checkbox disabled={true} checked={false} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render Checkbox correctly when disable, checked is true', () => {
    // when
    const { toJSON } = render(<Checkbox disabled checked={true} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render Checkbox correctly when disable, checked and indeterminate is true', () => {
    // when
    const { toJSON } = render(
      <Checkbox disabled checked={true} indeterminate={true} />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should press checkbox', () => {
    // given
    const onPress = jest.fn();

    // when
    const { getByTestId } = render(
      <Checkbox checked={true} testID="checkbox" onPress={onPress} />,
    );
    const checkboxComponent = getByTestId('checkbox');
    fireEvent.press(checkboxComponent);

    // then
    expect(onPress).toBeCalledTimes(1);
  });

  test('should not press checkbox when disabled', () => {
    // given
    const onPress = jest.fn();

    // when
    const { getByTestId } = render(
      <Checkbox
        checked={true}
        disabled
        testID="checkbox"
        label="lorem ipsum"
        onPress={onPress}
      />,
    );
    const checkboxComponent = getByTestId('checkbox');
    const checkboxLabelComponent = getByTestId('label');
    fireEvent.press(checkboxComponent);

    // then
    expect(onPress).not.toBeCalled();
    expect(checkboxLabelComponent.props.style[0].color).toBe(
      theme.colors.neutralLighter,
    );
  });

  test('should render without icon correctly', () => {
    // when
    const { queryByTestId } = render(<Checkbox checked={false} />);
    const checkboxCheckedIcon = queryByTestId('check-icon');
    const checkboxMinusIcon = queryByTestId('minus-icon');

    // then
    expect(checkboxCheckedIcon).toBeFalsy();
    expect(checkboxMinusIcon).toBeFalsy();
  });

  test('should render checked icon correctly', () => {
    // when
    const { getByTestId } = render(<Checkbox checked={true} />);
    const checkboxIcon = getByTestId('check-icon');

    // then
    expect(checkboxIcon).toBeTruthy();
  });

  test('should render unchecked icon correctly', () => {
    // when
    const { getByTestId } = render(
      <Checkbox checked={true} indeterminate={true} />,
    );
    const checkboxIcon = getByTestId('minus-icon');

    // then
    expect(checkboxIcon).toBeTruthy();
  });

  test('should render with label correctly when checked', () => {
    // when
    const { getByText } = render(
      <Checkbox checked={true} label="lorem ipsum" />,
    );
    const checkboxLabel = getByText('lorem ipsum');

    // then
    expect(checkboxLabel).toBeTruthy();
    expect(checkboxLabel.props.style[0].color).toBe(theme.colors.primaryKey);
  });

  test('should render with label correctly when unchecked', () => {
    // when
    const { getByText } = render(
      <Checkbox checked={false} label="lorem ipsum" />,
    );
    const checkboxLabel = getByText('lorem ipsum');

    // then
    expect(checkboxLabel).toBeTruthy();
    expect(checkboxLabel.props.style[0].color).toBe(theme.colors.neutralDarker);
  });

  test('should render without label correctly', () => {
    // when
    const { queryByTestId } = render(<Checkbox checked={true} />);
    const checkboxLabel = queryByTestId('');

    // then
    expect(checkboxLabel).toBeFalsy();
  });

  test('should render with icon correctly when checked', () => {
    // when
    const { getByTestId } = render(
      <Checkbox checked={true} flagIcon="turkey" />,
    );
    const icon = getByTestId('flag-icon');

    // then
    expect(icon).toBeTruthy();
  });

  test('should render with icon correctly when unchecked', () => {
    // when
    const { getByTestId } = render(
      <Checkbox checked={false} flagIcon="turkey" />,
    );
    const icon = getByTestId('flag-icon');

    // then
    expect(icon).toBeTruthy();
  });

  test('should render with icon correctly when disabled', () => {
    // when
    const { getByTestId } = render(
      <Checkbox checked={false} disabled flagIcon="turkey" />,
    );
    const icon = getByTestId('flag-icon');

    // then
    expect(icon).toBeTruthy();
  });

  test('should render without icon correctly when icon prop is not provided', () => {
    // when
    const { queryByTestId } = render(<Checkbox checked={false} />);
    const icon = queryByTestId('flag-icon');

    // then
    expect(icon).toBeFalsy();
  });
});
