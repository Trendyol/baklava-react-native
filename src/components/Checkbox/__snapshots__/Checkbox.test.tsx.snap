// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render Checkbox correctly when checked is false 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="tickbox"
    accessible={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#AFBBCA",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="tickbox"
  />
</View>
`;

exports[`Checkbox should render Checkbox correctly when checked is true 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="tickbox"
    accessible={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F27A1A",
            "borderColor": "#F27A1A",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="tickbox"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="10"
      bbWidth="10"
      fill="#FFFFFF"
      focusable={false}
      height={10}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 10,
            "width": 10,
          },
        ]
      }
      testID="check-icon"
      title="check"
      vbHeight={24}
      vbWidth={24}
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          Object {
            "type": 0,
            "value": 4294967295,
          }
        }
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M23.401 3.573a2 2 0 0 1 .026 2.828l-13.75 14a2 2 0 0 1-2.854 0l-6.25-6.363a2 2 0 0 1 2.854-2.803l4.823 4.91L20.573 3.6a2 2 0 0 1 2.828-.026Z"
          fill={
            Object {
              "type": 0,
              "value": 4294967295,
            }
          }
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Checkbox should render Checkbox correctly when checked is true, indeterminate is false 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="tickbox"
    accessible={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F27A1A",
            "borderColor": "#F27A1A",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="tickbox"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="10"
      bbWidth="10"
      fill="#FFFFFF"
      focusable={false}
      height={10}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 10,
            "width": 10,
          },
        ]
      }
      testID="check-icon"
      title="check"
      vbHeight={24}
      vbWidth={24}
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          Object {
            "type": 0,
            "value": 4294967295,
          }
        }
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M23.401 3.573a2 2 0 0 1 .026 2.828l-13.75 14a2 2 0 0 1-2.854 0l-6.25-6.363a2 2 0 0 1 2.854-2.803l4.823 4.91L20.573 3.6a2 2 0 0 1 2.828-.026Z"
          fill={
            Object {
              "type": 0,
              "value": 4294967295,
            }
          }
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Checkbox should render Checkbox correctly when checked is true, indeterminate is true 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="tickbox"
    accessible={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F27A1A",
            "borderColor": "#F27A1A",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="tickbox"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="8"
      bbWidth="8"
      fill="#FFFFFF"
      focusable={false}
      height={8}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 8,
            "width": 8,
          },
        ]
      }
      testID="minus-icon"
      title="minus"
      vbHeight={24}
      vbWidth={24}
      width={8}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          Object {
            "type": 0,
            "value": 4294967295,
          }
        }
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M0 12a2 2 0 0 1 2-2h20a2 2 0 1 1 0 4H2a2 2 0 0 1-2-2Z"
          fill={
            Object {
              "type": 0,
              "value": 4294967295,
            }
          }
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Checkbox should render Checkbox correctly when disable, checked and indeterminate is true 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="tickbox"
    accessible={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F1F2F7",
            "borderColor": "#F1F2F7",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="tickbox"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="8"
      bbWidth="8"
      fill="#95A1B5"
      focusable={false}
      height={8}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 8,
            "width": 8,
          },
        ]
      }
      testID="minus-icon"
      title="minus"
      vbHeight={24}
      vbWidth={24}
      width={8}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          Object {
            "type": 0,
            "value": 4287996341,
          }
        }
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M0 12a2 2 0 0 1 2-2h20a2 2 0 1 1 0 4H2a2 2 0 0 1-2-2Z"
          fill={
            Object {
              "type": 0,
              "value": 4287996341,
            }
          }
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Checkbox should render Checkbox correctly when disable, checked is false 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="tickbox"
    accessible={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F1F2F7",
            "borderColor": "#AFBBCA",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="tickbox"
  />
</View>
`;

exports[`Checkbox should render Checkbox correctly when disable, checked is true 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <View
    accessibilityLabel="tickbox"
    accessible={true}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F1F2F7",
            "borderColor": "#F1F2F7",
            "borderRadius": 2,
            "borderWidth": 1,
            "height": 16,
            "justifyContent": "center",
            "width": 16,
          },
        ],
      ]
    }
    testID="tickbox"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="10"
      bbWidth="10"
      fill="#95A1B5"
      focusable={false}
      height={10}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 10,
            "width": 10,
          },
        ]
      }
      testID="check-icon"
      title="check"
      vbHeight={24}
      vbWidth={24}
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={
          Object {
            "type": 0,
            "value": 4287996341,
          }
        }
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M23.401 3.573a2 2 0 0 1 .026 2.828l-13.75 14a2 2 0 0 1-2.854 0l-6.25-6.363a2 2 0 0 1 2.854-2.803l4.823 4.91L20.573 3.6a2 2 0 0 1 2.828-.026Z"
          fill={
            Object {
              "type": 0,
              "value": 4287996341,
            }
          }
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
